src/main/java/org/mitre/synthea/export/rif/RIFExporter.java

```diff
@@ -183,6 +182,43 @@ protected List<String> getDiagnosesCodes(Person person, long time) {
     return mappedDiagnosisCodes;
   }
 
+  /**
+   * Returns the earliest diagnosis time stamp for a particular condition.
+   * @param person patient with the diagnoses
+   * @param code the condition code
+   * @return the diagnosis time stamp or Long.MAX_VALUE if not diagnosed
+   */
+  protected long getEarliestDiagnosis(Person person, String code) {
+    List<HealthRecord.Entry> diagnoses = new ArrayList<HealthRecord.Entry>();
+    for (HealthRecord.Encounter encounter : person.record.encounters) {
+      for (HealthRecord.Entry dx : encounter.conditions) {
+        if (exporter.conditionCodeMapper.canMap(dx.codes.get(0).code)) {
+          String mapped = exporter.conditionCodeMapper.map(dx.codes.get(0).code, person, true);
+          if (mapped.equals(code)) {
+            diagnoses.add(dx);
+          }
+        }
+      }
+    }
+    if (!diagnoses.isEmpty()) {
+      // Sort them by date and then return the oldest
+      diagnoses.sort(ENTRY_SORTER);
+      return diagnoses.get(diagnoses.size() - 1).start;
+    }
+    return Long.MAX_VALUE;
+  }
+
+  /**
+   * Test whether a person has part A and B coverage at the specified timestamp.
+   * @param person the person
+   * @param timestamp the timestamp to check coverage at
+   * @return true if covered, false otherwise
+   */
+  public boolean hasPartABCoverage(Person person, long timestamp) {
+    Long coverageStartDate = (Long)person.attributes.get(RIFExporter.COVERAGE_START_DATE);
+    return timestamp >= coverageStartDate;
+  }
+
```

Comment:
I see why you did this, and it is fine for now, but we really should try to integrate this with the classes in the `org.mitre.synthea.world.concepts.healthinsurance` package.